src/bson_array.c

```diff
@@ -288,7 +288,7 @@ size_t bson_array_from_bytes_len(BsonArray *output, const uint8_t *data, size_t
     }
   }
 
-  if (parseError) {
+  if (parseError || dataSize - remainBytes == 0) {
```

Comment:
Hmmm, I'm not seeing any way that this could happen. Looking at the code, `dataSize - remainBytes` should always be greater than or equal to `5` at this point.

Reply:
This is definitely a false-positive, but it was decided to still include this addition to silence the warning